daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/JsonApiIt.scala

```diff
@@ -270,7 +271,11 @@ final class JsonApiIt
       )
       .toMap
     val participantParams = Participants(Some(defaultParticipant), participantMap, partyMap)
-    Runner.jsonClients(participantParams, envIface)
+    Runner.jsonClients(participantParams, envIface).andThen { case Success(ps) =>
+      partyMap.foreach { case (party, participant) =>
+        ps.participants.get(participant).map(_.allocateParty(party, ""))
+      }
+    }
```

Comment:
I am wondering if you might have some flaky tests because of this implementation here, as a result of the internal Future which allocates a party is not being waited, so if it is slow enough - you will have a problem

Reply:
You are absolutely right. Thanks for spotting that!